giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/aspects/access/RestSecurityAspect.java

```diff
@@ -225,6 +231,69 @@ public Object checkDocument(ProceedingJoinPoint joinPoint, DocumentAccessCheck c
     }
     
     @Around("within(edu.asu.diging.gilesecosystem.web.rest..*) && @annotation(check)")
+    public Object checkImageAccess(ProceedingJoinPoint joinPoint, ImageAccessCheck check ) throws Throwable {
+
+        UserTokenObject userTokenObj = extractUserTokenInfo(joinPoint, check.value(), null);
+        User user = userTokenObj.user;
+        String token = userTokenObj.token;
+
+        if (user == null) {
+            throw new AspectMisconfigurationException(
+                    "User object is missing in method.");
+        }
+        
+        // Get the Method signature and the arguments passed to the method.
+        MethodSignature methodSignature = (MethodSignature) joinPoint.getSignature();
+        Method method = methodSignature.getMethod();
+        Parameter[] paras = method.getParameters();
+        
+        int imagePathParamIdx = -1;
+        
+        // Get all arguments that are passed to the calling method.
+        Object[] arguments = joinPoint.getArgs();
+        
+        if (paras != null) {
+
+            for (int i = 0; i < paras.length; i++) {
+                Parameter p = paras[i];
+                if (p.getAnnotation(InjectImagePath.class) != null) {
+                    imagePathParamIdx = i;
+                }
+            }
+
+            if (imagePathParamIdx != -1) {
+
+                String imagePath = extractImagePath(joinPoint);
```

Comment:
image path parameter is where I need to assign image path to send in to main class. I do not get path directly from parameters here, need to read joinpoint arguments and find path starting with "fn=...."